sql/src/main/java/io/crate/planner/PlanPrinter.java

```diff
@@ -244,4 +247,25 @@ private ExecutionPhase2MapVisitor() {
                 .put("mergePhase", phaseMap(countPlan.mergePhase()));
         }
     }
+
+    /**
+     * Converts the shardId's of each node->table from a {@link IntIndexedContainer} to a list of integers as hppc
+     * primitive structures are not supported by the {@link org.elasticsearch.common.xcontent.XContentBuilder}.
+     */
+    private static Map<String, Map<String, List<Integer>>> xContentSafeRoutingLocations(
+        Map<String, Map<String, IntIndexedContainer>> locations) {
+        HashMap<String, Map<String, List<Integer>>> safeLocations = new HashMap<>(locations.size());
```

Comment:
minor: since we know the expected number of elements we should divide it by the load factor (default .75) to avoid map rehashing (same below)